src/jit/flowgraph.cpp

```diff
@@ -20705,6 +20726,10 @@ void Compiler::fgDebugCheckFlags(GenTreePtr tree)
                 chkFlags |= GTF_ORDER_SIDEEFF;
                 break;
 
+            case GT_MEMORYBARRIER:
+                chkFlags |= GTF_GLOB_REF | GTF_ASG;
```

Comment:
I'm a bit surprised that `GT_MEMORYBARRIER` does not imply `GT_ORDER_SIDEEFF`...

Reply:
I'm surprised by that too but the meaning of GTF_ORDER_SIDEEFF and GTF_GLOB_REF is not very well defined at the moment so I deferred the work to fix and check them and just kept the existing behavior.

Reply:
`GTF_ORDER_SIDEEFF` is well defined: it is a full compiler barrier. I suppose that when you say that its meaning is not well defined you're referring to the fact that we don't always check it?

Reply:
I don't think it's currently always used that way. In any case, checking GTF_ORDER_)SIDEFF is a future work item and I'll investigate why it's not set on GT_MEMORYBARRIER at that time.